### DISCUSSION TITLE
workflow for updating large_media

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#22611] [Q&A General] [answered]
Hi everyone,

In a PR of mine, I've got some pics that need to be placed into large_media.  I've obviously mucked something up with git, as i get: "Failed to merge submodule large_media (not checked out)" from the CIVET precheck.   Can someone help please?

 I looked at the related discussion, #18387 , but that doesn't seem relevant here.

Here's roughly what i did:

```
cd ~/projects/moose/large_media
git checkout master
git pull
cd ~/projects/moose
git submodule update large_media
cd ~/projects/moose/large_media/geochemistry
git add <my files>
git commit -m "geoechemistry pics added.  Refs #21592"
git push
cd ~/projects/moose
git add large_media
git commit -m "large media updated with biogeochemistry pics"
git push origin biogeochem_21559
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

What does `git show` say on this commit?

Guillaume

**Reply by WilkAndy** [reply]
```
> git show

commit e1cade2cf70ca49d09ba6a529f2e63ae0d88c207 (HEAD -> biogeo_21559, origin/biogeo_21559)
Author: Andy Wilkins <andrew.wilkins@csiro.au>
Date:   Mon Aug 1 14:11:06 2022 +1000

    doc and tests and cpgr changes
    
    also: large_media updated with biogeochemistry pics
    
    Refs #21559

diff --git a/large_media b/large_media
index e1211aaf2d..50849f0697 160000
--- a/large_media
+++ b/large_media
@@ -1 +1 @@
-Subproject commit e1211aaf2dec766e9a8e2da1631ee2f36ec7c5de
+Subproject commit 50849f069720666d8979e0083bd19dea5dc69e11
...
```

**Reply by WilkAndy** [reply]
Please could you help, @GiudGiud .   I just removed the "git add large_media" from my PR ( #21592 ) and now the precheck of CIVET passes.  However, CIVET is pulling the wrong commit of large_media:

```
BUILD_ROOT/moose/: git submodule update --init --checkout --reference /data/gitrefs/github.com/idaholab/large_media large_media
Submodule 'large_media' (https://github.com/idaholab/large_media) registered for path 'large_media'
Cloning into '/data/civet3/build/moose/large_media'...
Submodule path 'large_media': checked out '88bc8647f9f9954554485d010d7a6311e20fca71'
```

it should be checking out `50849f069720666d8979e0083bd19dea5dc69e11` , which is the latest commit to large_media that includes my additions.  This means that my PR will probably eventually fail because the documentation website will not build correctly.

**Reply by GiudGiud** [reply]
ok I just tried the same thing you did. 

If that doesnt work, I d rebase your PR on current moose/devel.

**Reply by WilkAndy** [reply]
I believe that your PR with the updated large_media submodule failed in the same way as mine did (above)   Are you sure that it isn't something to do with 

```
git submodule update --init --checkout --reference /data/gitrefs/github.com/idaholab/large_media large_media
```

somehow not getting my latest commit into `large_media` ?

**Reply by WilkAndy** [reply]
@GiudGiud , i believe your attempt above failed.  So, i have now done the following, after all my MOOSE commits

```
git fetch upstream
git rebase upstream/next
git add large_media
git commit -m "Andy performed git add large_media after rebasing from upstream/next"
git push -f origin biogeo_21559
```

**Reply by GiudGiud** [reply]
yeah that was the thing to do

the line that mattered and that I missed
CONFLICT (submodule): Merge conflict in large_media

it could not merge your PR into devel because of the difference in submodules

**Reply by WilkAndy** [reply]
Thank you @GiudGiud .   It has passed the precheck.  I will close this discussion if it passes all tests.

**Reply by GiudGiud** [reply]
btw you should rebase off of devel not next. You wont get coverage diffs off of next

**Reply by WilkAndy** [reply]
The PR now passes.  In conclusion:

- the workflow presented in my initial question was almost correct
- At some stage, for example before the `git add large_media`, i should have `git fetch upstream ; git rebase upstream/next`

**Reply by WilkAndy** [reply]
Thanks for working with me on this, @GiudGiud
